common/beerocks/bwl/common/key_value_parser.cpp

```diff
@@ -0,0 +1,225 @@
+
+#include <bwl/key_value_parser.h>
+
+#include <bcl/beerocks_state_machine.h>
+#include <bcl/beerocks_string_utils.h>
+#include <bcl/network/network_utils.h>
+
+#include <easylogging++.h>
+#include <sstream>
+
+using namespace bwl;
```

Comment:
Why do you hate 'using namespace' so much?

Reply:
> Why do you hate 'using namespace' so much?

This particular one is confusing: everything we are *defining* in this file *is* actually in the bwl namespace. I think it's a lot clearer to have
```
namespace foo {
void xxx() {} // This defines foo::xxx
}
```
than
```
using namespace foo;

void xxx() {} // This defines ::xxx() unless there was a preceding declaration of foo::xxx()
```

Second, in general I don't like `using namespace` because it basically bypasses the namespace. Namespaces are meant to split things up in modules so you can clearly see to which module each thing belongs. `using namespace` throws them back into a single heap, so then why did you put them into two separate namespaces to begin with?

I don't consider it that important, but I *will* consistently comment about it :-)

Reply:
Since it does not matter to be that much. I'll change it to how you want. 🙂